src/coreclr/vm/pendingload.cpp

```diff
@@ -12,241 +12,365 @@
 
 #ifndef DACCESS_COMPILE
 
+#ifdef PENDING_TYPE_LOAD_TABLE_STATS
+// Enable PENDING_TYPE_LOAD_TABLE_STATS to gather counts of Entry structures which are allocated
 
-// ============================================================================
-// Pending type load hash table methods
-// ============================================================================
-/*static */ PendingTypeLoadTable* PendingTypeLoadTable::Create(LoaderHeap *pHeap,
-                                                               DWORD dwNumBuckets,
-                                                               AllocMemTracker *pamTracker)
+static LONG pendingTypeLoadEntryDynamicAllocations = 0;
+void PendingTypeLoadEntryDynamicAlloc()
 {
-    CONTRACTL
+    InterlockedIncrement(&pendingTypeLoadEntryDynamicAllocations);
+}
+#endif // PENDING_TYPE_LOAD_TABLE_STATS
+
+PendingTypeLoadTable PendingTypeLoadTable::s_table;
+
+PendingTypeLoadTable::Entry::Entry()
+    : m_typeKey(TypeKey::InvalidTypeKey()),
+        m_fIsPreallocated(true)
+{
+}
+
+PendingTypeLoadTable::Entry::Entry(const TypeKey& typeKey)
+    : m_typeKey(typeKey),
+        m_fIsPreallocated(false)
+{
+#ifdef PENDING_TYPE_LOAD_TABLE_STATS
+    PendingTypeLoadEntryDynamicAlloc();
+#endif // PENDING_TYPE_LOAD_TABLE_STATS
+}
+
+void PendingTypeLoadTable::Entry::SetTypeKey(const TypeKey& typeKey)
```

Comment:
```suggestion
void PendingTypeLoadTable::Entry::SetTypeKey(TypeKey typeKey)
```

This should be okay.